opentelemetry-sdk/src/opentelemetry/sdk/_metrics/measurement_consumer.py

```diff
@@ -41,15 +46,31 @@ def collect(
 
 
 class SynchronousMeasurementConsumer(MeasurementConsumer):
+    def __init__(self, sdk_config: SdkConfiguration) -> None:
+        self._lock = Lock()
+        self._sdk_config = sdk_config
+        # should never be mutated
+        self._reader_to_state: Mapping[MetricReader, MetricReaderStorage] = {
+            reader: MetricReaderStorage(sdk_config)
+            for reader in sdk_config.metric_readers
+        }
+        self._async_instruments: List["_Asynchronous"] = []
+
     def consume_measurement(self, measurement: Measurement) -> None:
-        pass
+        for reader_state in self._reader_to_state.values():
```

Comment:
I might be missing something because the `reader_storage` and `View` have not been implemented yet. Is the reason we iterate through all readers when a measurement event happens (like add) because we need to find the relevant `View` to apply aggregation to?

Reply:
Each Reader will have have its own separate running aggregations for each view. All the Readers are independent from one another meaning they can each collect over different time intervals. See [this part of the spec](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/metrics/sdk.md#metricreader)

> The SDK MUST support multiple `MetricReader` instances to be registered on the same `MeterProvider`, and the [MetricReader.Collect](#collect) invocation on one `MetricReader` instance SHOULD NOT introduce side-effects to other `MetricReader` instances. For example, if a `MetricReader` instance is receiving metric data points that have [delta temporality](./datamodel.md#temporality), it is expected that SDK will update the time range - e.g. from (T<sub>n</sub>, T<sub>n+1</sub>] to (T<sub>n+1</sub>, T<sub>n+2</sub>] - **ONLY** for this particular `MetricReader` instance.

We could probably optimize this a bit, but there shouldn't be many readers/exporters in most cases.